dkulib/parallelizer/parallelizer.py

```diff
@@ -107,10 +137,10 @@ def __init__(
             raise ValueError("Please set at least one exception in exceptions_to_catch")
         self.parallel_workers = parallel_workers
         self.batch_support = batch_support
+        if batch_support is False:
```

Comment:
We should also make sure that if batch_support is False, then batch_response_parser is the default one

Reply:
assert are usually for unit tests, better to raise a ValueError

Reply:
After thinking again about this, wouldn't it be simpler to override batch_response_parser when batch_support is False? We do override batch_size anyway

Reply:
good idea! Updated it & added a comment

Reply:
Also
```suggestion
        if not batch_support:
```